src/indexer.h

```diff
@@ -516,6 +516,18 @@ struct FileContents {
   FileContents(const std::string& path, const std::string& content);
 };
 
+struct FileContentsWithOffsets {
+  std::string contents_;
+  // {0, 1 + position of first newline, 1 + position of second newline, ...}
+  std::vector<int> line_offsets_;
+
+  FileContentsWithOffsets();
+  FileContentsWithOffsets(std::string s);
+
+  optional<int> ToOffset(Position p) const;
+  std::string ContentsInRange(Range range) const;
```

Comment:
It'd be good to have a simple sanity test for this class. It should only take a minute or two to add/write.

Reply:
Where to put them?